public/pages/DetectorDetail/containers/DetectorDetail.tsx

```diff
@@ -365,6 +393,61 @@ export const DetectorDetail = (props: DetectorDetailProps) => {
               />
             </EuiFlexItem>
           </EuiFlexGroup>
+          {isResultIndexMissing || isCreatingIndex ? (
+            <EuiCallOut
+              style={{
+                marginLeft: '10px',
+                marginRight: '10px',
+                marginBottom: '10px',
+              }}
+              title={`Result index '${get(
+                detector,
+                'resultIndex',
+                ''
+              )}' has been deleted and all anomaly results have been lost. To start saving anomaly results, 
+              restart real-time or historical detection, or recreate the index.`}
+              color="danger"
+              iconType="alert"
+            >
+              <EuiButton
+                style={{ marginLeft: '24px' }}
+                fill={false}
+                isLoading={isCreatingIndex}
+                onClick={async () => {
+                  setIsCreatingIndex(true);
+                  const indexConfig = {
+                    index: get(detector, 'resultIndex', ''),
+                    body: {},
+                  };
+                  await dispatch(createIndex(indexConfig, true))
```

Comment:
Will we use default AD result index mapping to re-create custom result index?

Reply:
Currently it's pulling the mappings from [here](https://raw.githubusercontent.com/opensearch-project/anomaly-detection/main/src/main/resources/mappings/anomaly-results.json). More details on that [here](https://github.com/ohltyler/anomaly-detection-dashboards-plugin-1/blob/result-index-handling/server/routes/opensearch.ts#L185).

Reply:
What does it mean by fetching the latest anomaly result index mapping? Is this different than the initial default mapping? Also what would the name be of the recreated custom result index.

Reply:
It would be the default mapping based on the current version. But since the mappings have changed over different versions (like adding high-cardinality-related fields to the results), the idea is to get the corresponding version of the mappings. The name would be the stored custom result index for the detector.

Either way this functionality will probably be removed and we will let the backend init the index with the correct mappings.

Reply:
This is removed now.